### PR TITLE
[SPARK-38713][SQL] Change spark.sessionstate.conf.getConf/setConf operation to spark.conf.get/set

### PR DESCRIPTION
**Pull request by jackylee-ch**
### Why are the changes needed?
In the sql module, we provide `SparkSession.conf` as a unified entry for `SQLConf.set/get`, which can prevent users or logic from modifying StaticSQLConf and Spark configs. However, I found `SparkSession.sessionstate.conf` is used in some code to getConf or setConf, which can skip the check of `RuntimeConfig`.
In this PR, we want to unify the behavior of `SQLConf.getConf/setConf` to `SparkSession.conf`.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Origin GA

### CODE REVIEW

**Comment by jackylee-ch** [comment]
cc @srowen @cloud-fan @HyukjinKwon

**Comment by srowen** [comment]
I don't know this part well, but it seems reasonable to unify the checks here

File: sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala [hunk 1]
```diff
@@ -60,6 +60,16 @@ class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
     set(key, value.toString)
   }
 
+  /**
+   * Sets the given Spark runtime configuration property.
+   *
+   * @since 3.4.0
+   */
+  def set[T](entry: ConfigEntry[T], value: T): Unit = {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala:68]: `ConfigEntry` is a private class, that's why `protected[sql] def get[T](entry: ConfigEntry[T]): T` is not public. I think we should do the same for `set`.

Comment (jackylee-ch) [reply sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala:68]: Sure, I will change it.

**Comment by jackylee-ch** [comment]
@cloud-fan @HyukjinKwon Any more question about this pr?
BTW, can we remove the [AmplabJenkins](https://github.com/AmplabJenkins) report Since we won't use Jenkins any more?

File: sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala [hunk 2]
```diff
@@ -60,6 +60,16 @@ class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
     set(key, value.toString)
   }
 
+  /**
+   * Sets the given Spark runtime configuration property.
+   *
+   * @since 3.4.0
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala:66]: non-public API doesn't need the since annotation.

Comment (jackylee-ch) [reply sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala:66]: okey

**Review by cloud-fan — APPROVED** [review]

**Comment by jackylee-ch** [comment]
cc @srowen @cloud-fan @HyukjinKwon

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by cloud-fan** [comment]
@jackylee-ch seems you link to the wrong JIRA ticket?

**Comment by jackylee-ch** [comment]
> @jackylee-ch seems you link to the wrong JIRA ticket?

This is a minor change, thus I didn't create a JIRA for it.

**Comment by cloud-fan** [comment]
At least we shouldn't link to an unrelated JIRA... Please create a new one for this PR as it's not a minor one-line change.

**Comment by jackylee-ch** [comment]
> At least we shouldn't link to an unrelated JIRA... Please create a new one for this PR as it's not a minor one-line change.

Sorry, I will create one inmediatly.

**Comment by jackylee-ch** [comment]
https://issues.apache.org/jira/browse/SPARK-38713
This is the new Created Jira. @cloud-fan

**Pull request closed without merging**